core/src/main/java/org/jdbi/v3/core/internal/JdbiClassUtils.java

```diff
@@ -56,4 +67,11 @@ public static <T> T createInstance(Class<T> factoryClass) {
             throw new IllegalStateException(format("Unable to instantiate class %s", factoryClass), e);
         }
     }
+
+    public static Stream<Class<?>> superTypes(Class<?> type) {
+        Class<?>[] interfaces = type.getInterfaces();
```

Comment:
Should this helper method emit duplicate super types? Like if A implements B,C and B implements C, you would emit C twice. (I realize this is not a new problem, but if we provide a helper, maybe we should specify it one way or the other.)

Reply:
I added a deduplication step.